packages/net-stubbing/lib/server/intercepted-request.ts

```diff
@@ -184,7 +184,31 @@ export class InterceptedRequest {
           }
         }
 
-        if (eventName === 'before:request') {
+        if (
+          eventName === 'before:request' ||
+          // https://github.com/cypress-io/cypress/issues/17139
+          // This `cypressInternalResReplyCall` came
+          // from driver/src/cy/net-stubbing/events/before-requests.ts
+          // to sendStaticResponse at util.ts
+          // to middleware/request.ts
+          // to middleware/response.ts
+          // to here.
+          //
+          // This is a really ugly solution.
+          // But `cypressInternalResReplyCall` is necessary below because of the 3 reasons.
```

Comment:
😁

Would it be easier to fix this bug by updating how `req.reply`/`res.send` are handled in the driver code? Currently, calls to `sendStaticResponse` dispatch a single `send:static:response` event:

https://github.com/cypress-io/cypress/blob/636aae03b0b504635d2ccb013cbc7ae9b2ce735e/packages/driver/src/cy/net-stubbing/events/before-request.ts#L277-L277

https://github.com/cypress-io/cypress/blob/636aae03b0b504635d2ccb013cbc7ae9b2ce735e/packages/driver/src/cy/net-stubbing/events/response.ts#L80-L80

https://github.com/cypress-io/cypress/blob/636aae03b0b504635d2ccb013cbc7ae9b2ce735e/packages/driver/src/cy/net-stubbing/events/index.ts#L56-L61

But... it seems like this whole bug that this PR fixes, is being caused by `send:static:response` not ultimately resolving the Promise in `handleSubscriptions`:

https://github.com/cypress-io/cypress/blob/636aae03b0b504635d2ccb013cbc7ae9b2ce735e/packages/net-stubbing/lib/server/intercepted-request.ts#L164-L166

So maybe a more elegant fix would be to get rid of `send:static:response` and update `event:handler:resolved` to add a `staticResponse` field. What do you think @sainthkh ?

Reply:
I tried other solutions like creating a record that tracks request ID and route ID. But it doesn't work, either. 

I'll try yours tomorrow.